src/Input.js

```diff
@@ -6,23 +6,36 @@ import uuid from 'node-uuid';
 
 class Input extends React.Component {
   componentDidMount() {
-    if (this.props.type === 'select' && !this.props.browserDefault && typeof $ !== 'undefined') {
-      $(ReactDOM.findDOMNode(this.refs.inputEl)).material_select();
-    }
+    this._setupEl();
   }
-
   componentDidUpdate() {
-    if (this.props.type === 'select' && !this.props.browserDefault && typeof $ !== 'undefined') {
-      var $el = $(ReactDOM.findDOMNode(this.refs.inputEl));
-      $el.material_select('destroy');
-      $el.material_select();
+    this._setupEl();
+  }
+
+  _setupEl() {
+    if (typeof $ !== 'undefined') {
+      var $inputEl = $(ReactDOM.findDOMNode(this.refs.inputEl));
+
+      if (this.props.type === 'select' && !this.props.browserDefault) {
+        $inputEl.material_select('destroy');
+        $inputEl.material_select();
+      }
+      else {
+        var $labelEl = $(ReactDOM.findDOMNode(this.refs.labelEl));
+        if ($inputEl.val()) {
+          $labelEl.addClass('active');
```

Comment:
@isaiah I tried to use `classnames` for this one too like I did with the rest, but it's not straightforward.

An `input` can get its value from different props like `value` or `valueLink` or even `defaultValue`. The only clean way I found to figure out for sure if an `input` has a value is to do it programatically after the component mounts.

Any suggestions on how to tackle this one would be appreciated :)